### PR TITLE
fix: active header navigation item indicator

### PR DESCRIPTION
**Pull request by nikomakela**
LIIKUNTA-617

### Frontpage - no item selected
image

### Search view
/fi/search
image
/fi/haku
image
/sv/sok
image

### Lapset ja perheet
/fi/pages/kohderyhmat/lapset-ja-perheet
image
/fi/sivut/kohderyhmat/lapset-ja-perheet
image

### Kohderyhmät
/fi/pages/kohderyhmat
image
/fi/sivut/kohderyhmat
image
/sv/sidor/malgrupper
image
/sv/pages/malgrupper
image

### CODE REVIEW

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr639.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr639.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr639.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr639.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr639.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr639.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr639.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr639.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr639.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr639.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr639.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr639.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr639.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr639.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by nikomakela** [comment]
I tried, but I could not find a way to unit test these i18n routes that are implemented with next js router and redirect configuration.

- [ ] Fix the unit tests

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr639.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr639.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr639.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr639.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr639.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr639.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by karisal-anders** [comment]
It seems the hierarchy doesn't come from the hierarchy of links in the menu but somehow from their URL parts? I mean testing locally by selecting "Liikuntatilojen varaaminen" in the Finnish menu I get both "Kohderyhmät" and "Liikuntatilojen varaaminen" as selected because "Kohderyhmät" = /pages/kohderyhmat is a parent URL in a breadcrumb sense of "Liikuntatilojen varaaminen" i.e. /pages/kohderyhmat/kumppanit/tilojen-varaaminen/:

image

Just an observation and something that needs to be understood when creating the menus so they work correctly. Perhaps sharing this information with the product owner would be good so they know that they have to create the menus in a breadcrumb sense hierarchically for this functionality to work correctly i.e. they can't put just anything under a menu and expect for this to work unless we make this work using the menus' hierarchy of items and not using the breadcrumb hierarchy of the URLs themselves.

**Comment by karisal-anders** [comment]
What about after selecting an article, should the article URL in the menu still show as selected? Example:

Having clicked on the "Topical" menu entry in the English menu -> http://localhost:3000/en/articles
 - image

Clicking on an article on the above page -> http://localhost:3000/en/articles/uncategorized-en/numerous-outdoor-gyms-around-the-city and although the URL begins with the /en/articles part which matches the "Topical" menu entry the "Topical" menu entry is not shown as selected:
image

So, what do you think? Should the menu entry be selected in this case?

**Comment by karisal-anders** [comment]
## Test results using **production** data:

### Finnish production menu

In the production **Finnish** menu selecting any of the menu links at any level worked as expected.

### Swedish production menu

In the production **Swedish** menu everything else worked as expected except when selecting any of the target group menu's second level entries, e.g. `Barn och familjer` the first level entry `Målgrupper` always remained unselected:
- image

### English production menu

In the production English menu everything worked as in the Swedish menu, i.e. the target group subselections left the "Target group" top level entry unselected:
- image

### Summary

For comparison here's the Finnish menu with "Lapset ja aikuiset" target group selected:
- image

So to summarize there's a **discrepancy** **between** the **languages** here, Finnish handles the target groups differently than Swedish or English.

**Comment by nikomakela** [comment]
@karisal-anders I think the issues should be fixed now. I would say that this somewhat fixes the issue that we had earlier. At least it makes the situation much better. However, I would say that only way to fully fix this, would be delivering some state-setter in the HCRC-component from the parent item to the child item, so the child could trigger the parent when it is self activated. But that is a bigger change and should be rerefined.

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr639.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr639.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr639.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr639.dev.hel.ninja :laughing::tada::tada::tada:

**Review by karisal-anders — APPROVED** [review]

**Comment by karisal-anders** [comment]
Tested with production data, all the menu items worked as expected.
